lib/rubocop/cop/performance/detect.rb

```diff
@@ -70,6 +73,12 @@ def preferred_method
           config.for_cop('Style/CollectionMethods') \
             ['PreferredMethods']['detect'] || 'detect'
         end
+
+        def lazy?(node)
+          return false if node.nil?
+          receiver, method, _args = *node
+          method == :lazy && !receiver.nil?
```

Comment:
This would affect the result of the predicate - it's best return only boolean values from predicates. We should write a cop for this one of those days. ;-)